### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by devendra-alt**
### 👨‍💻 requirements fulfilled
✔️ Created `nameable class interface`
✔️ Made `person` class `nameable`
✔️ created base `decorator`
✔️ Added `trimmer_decorator` for nameable object
✔️ Added `capitalize_decorator` for nameable object
✔️ Performed operations on the nameable person object

### CODE REVIEW

File: school/decorator.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative 'nameable'
+class Decorator < NameAble
+  def initialize(nameable)
```
Comment (usorfaitheloho) [inline school/decorator.rb:5]: please kindly add an ` attr_accessor: nameable` for the initialization of the decorator 😄

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @devendra-alt  ,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  [ ](https://camo.githubusercontent.com/679757134a8fee76cfe1941f9f3ecb0e0cf56dd02da4498268ce31f0dacdfca3/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)
> 
> ## Highlights
> * Decorator classes were written✔️

> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: school/decorator.rb (hunk 1 above)
Comment (devendra-alt) [reply school/decorator.rb:5]: thank you @usorfaitheloho  for valuable feedback 💯

**Review by Munsa1 — APPROVED** [review]
Hi @devendra-alt, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**